### PR TITLE
fix: fix deploy  for Nelp in requirements installation.

### PR DESCRIPTION
**Pull request by johanv26**
# fix: fix deploy with several retries and comments for Nelp.

## Description. 
These commits are a solution by the index proposed in
commit 02cc57ae0ee0881df01883a6a88c94d49a3480a5

This solve problem when the mongo db has key errors or problem with indexing. 

```

    self._command(\", \"  File \\\"/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/pymongo/collection.py\\\", line 235, in _command\", \"    return sock_info.command(\", \"  File \\\"/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/pymongo/pool.py\\\", line 603, in command\", \"    return command(self.sock, dbname, spec, slave_ok,\", \"  File \\\"/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/pymongo/network.py\\\", line 165, in command\", \"    helpers._check_command_response(\", \"  File \\\"/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/pymongo/helpers.py\\\", line 152, in _check_command_response\", \"    raise DuplicateKeyError(errmsg, code, response)\", \"pymongo.errors.DuplicateKeyError: E11000 duplicate key error collection: edxapp.modulestore.active_versions index: org_1_course_1_run_1 dup key: { : \\\"Demo\\\", : \\\"Tfx_S\\\", : \\\"2021\\\" }\

```
These other commits also solve other problems.
[fix: fix not install deps with extra requirements.](https://github.com/eduNEXT/configuration/pull/4/commits/56b81757f60d56f41b97b4a9426c9c4bd62192f0)[](https://github.com/johanv26) 

 

[fix: retries by node packages failing sometimes.](https://github.com/eduNEXT/configuration/pull/4/commits/ca0ebaffee9a86fa670e93796fe89d7d67ecdaf8) 

This by the error of backbone-relational in services like
ecommerce. EINVRES failed with 522 error code related to cloudfare.

### CODE REVIEW

**Comment by johanv26** [comment]
@MoisesGSalas How do you see it?

File: playbooks/roles/edx_django_service/tasks/main.yml [hunk 1]
```diff
@@ -184,7 +187,7 @@
   pip:
     name: "{{ item.name }}"
     version: "{{ item.version|default(omit) }}"
-    extra_args: "--exists-action w {{ item.extra_args|default('') }}"
+    extra_args: "--no-deps --exists-action w {{ item.extra_args|default('') }}"
```
Comment (MoisesGSalas) [inline playbooks/roles/edx_django_service/tasks/main.yml:190]: `--no-deps` may break de installation of any package that is added to `{{ <DJANGO_SERVICE>_EXTRA_REQUIREMENTS }}`

**Comment by johanv26** [comment]
@MoisesGSalas You are right, I quit no deps argument in extra requirements installation commit.

**Review by felipemontoya — APPROVED** [review]
This is working and it looks good to me for a customer branch

**Pull request merged**